utils/modeler/errors.go

```diff
@@ -0,0 +1,61 @@
+package modeler
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// NilLiteralModelError represents a failed attempt to populate a "model" because a nil literal was
+// passed to the modeler instead of a struct pointer.
+type NilLiteralModelError struct {
+}
+
+func newNilLiteralModelError() NilLiteralModelError {
+	return NilLiteralModelError{}
+}
+
+func (e NilLiteralModelError) Error() string {
+	return "Cannot populate a nil literal from the map."
+}
+
+// NonPointerModelError represents a failed attempt to populate a model because something other
+// than a pointer was passed to the modeler.
+type NonPointerModelError struct {
+	tipe reflect.Type
+}
+
+func newNonPointerModelError(tipe reflect.Type) NonPointerModelError {
+	return NonPointerModelError{tipe: tipe}
+}
+
+func (e NonPointerModelError) Error() string {
+	return fmt.Sprintf("Cannot populate non-pointer type %s from the map.", e.tipe)
+}
+
+// NilModelError represents a failed attempt to populate a "model" because a nil pointer was
+// passed to the modeler instead of a struct pointer.
+type NilModelError struct {
+	tipe reflect.Type
+}
+
+func newNilModelError(tipe reflect.Type) NilModelError {
```

Comment:
What if my intention is that nothing outside the modeler package is creating errors or these types?  Exporting them is largely a courtesy because working with unexported types can be a pain, but I try to remain defensive against instantiation of any type by anyone/anything that shouldn't be doing so.